---
title: "DESeq-WAFvsWF"
output: html_document
date: "2025-04-02"
---
```{r}
rm(list=ls())
```

```{r}
library(readr)

# Read the counts file
count_matrix <- read.csv('/rds/projects/f/foyerch-rnaseq-data-faba-bean/PCA/counts_matrix.csv')

# Extract and reorder column names
col_order <- colnames(count_matrix)

sorted_cols <- c(
  "Gene",
  col_order[-1][order(as.numeric(gsub("JC_(\\d+)_.*", "\\1", col_order[-1])))]
)

#remove zeroes
# Remove genes (rows) with more than 20 zeroes
count_matrix <- count_matrix[rowSums(count_matrix >1) >= 3, ]

count_matrix <- count_matrix[, sorted_cols]
```

```{r}
# Assign descriptive group labels
colnames(count_matrix) <- c(
  "Gene",
  rep("Wizard Control", 3),
  rep("Hiverna Control", 3),
  rep("Wizard & Freezing", 3),
  rep("Hiverna & Freezing", 3),
  rep("Wizard Acclimated", 3),
  rep("Hiverna Acclimated", 3),
  rep("Wizard Acclimated & Freezing", 3),
  rep("Hiverna Acclimated & Freezing", 3)
)

# Make column names unique (e.g., Wizard Acclimated.1, .2, .3)
colnames(count_matrix) <- make.unique(colnames(count_matrix))

```

```{r}
# Select only the relevant columns
selected_col <- grep("Gene|Wizard & Freezing|Wizard Acclimated & Freezing", colnames(count_matrix), value = TRUE)
WFvsWAF <- count_matrix[, selected_col]
```

```{r}
# Prepare DESeq2 count matrix
counts_matrix_WFvsWAF <- WFvsWAF
rownames(counts_matrix_WFvsWAF) <- counts_matrix_WFvsWAF$Gene
counts_matrix_WFvsWAF <- counts_matrix_WFvsWAF[, -1]  # remove Gene column

```

```{r}
sample_names <- colnames(counts_matrix_WFvsWAF)

# Define conditions in the correct order
conditions <- rep(c("Wizard Freezing", "Wizard Acclimated Freezing"), each = 3)
replicates <- rep(1:3, times = 2)

metadata <- data.frame(
  SampleID = sample_names,
  Condition = conditions,
  Replicate = as.factor(replicates)
)
rownames(metadata) <- sample_names

print(metadata)
```

#### Running DeSeq2 ####
```{r}
#### Running DeSeq2 ####
library(DESeq2)

# Create DESeq2 dataset with replicates as a factor
dds <- DESeqDataSetFromMatrix(countData = counts_matrix_WFvsWAF,
                              colData = metadata,
                              design = ~ Condition + Replicate)  # Include Replicate in design

# Run DESeq2
dds <- DESeq(dds)

# Extract results 
# pos log fold change is upregulated in acclimated & freezing
# neg log fold change is upregulated in freezing only

res <- results(dds, contrast = c("Condition", "Wizard Acclimated Freezing", "Wizard Freezing"))

# View results
head(res)
```

save ranked list of genes
```{r}
res <- res[!is.na(res$padj), ]
res <- res[res$padj < 0.05, ]
res$rank_score <- res$log2FoldChange
res <- res[order(-res$rank_score), ]

res_final <- data.frame(
  gene = rownames(res),
  rank = seq(1, nrow(res)),
  log2FoldChange = res$log2FoldChange,
  padj = res$padj
)

write.csv(res_final, "rankedlist_WFvsWAF.csv", row.names = FALSE)
```


#### Identifying Upregulated and Downregulated Genes ####
```{r}
library(dplyr)
library(tidyverse)

#add a column for significance based on thresholds:
res$Expression <- "NS"  # NS = Non-significant
res$Expression[res$padj < 0.001 & res$log2FoldChange > 4] <- "Upregulated" #upregulated genes
res$Expression[res$padj < 0.001 & res$log2FoldChange < -4] <- "Downregulated" #downregulated genes

#dataframe of upregulated genes
upregulated <- res %>%
  data.frame() %>% #convert results table to dataframe
  arrange(desc(log2FoldChange)) %>% #sort by log2FoldChange in descending order
  filter(Expression == "Upregulated") #select only the upregulated genes 

#top upregulated genes
head(upregulated)
```

```{r}
#dataframe of downregulated genes
downregulated <- res %>%
  data.frame() %>% #convert results table to dataframe
  arrange(log2FoldChange) %>% #sort by log2FoldChange in descending order
  filter(Expression == "Downregulated") #select only the upregulated genes 

#top downregulated genes
head(downregulated)
```

```{r}
#number of upregulated genes
up_genes <- nrow(upregulated)
#number of downregulated genes
down_genes <- nrow(downregulated)

#table of number of upregulated/downregulated genes
siggenes_no <- data.frame(upregulated_genes = up_genes, downregulated_genes = down_genes)
rownames(siggenes_no) <- "Number of differentially expressed genes"

head(siggenes_no)
```

```{r}
top_upregulated <- upregulated %>%
  arrange(padj) %>%
  slice_head(n = 5)

top_downregulated <- downregulated %>%
  arrange(padj) %>%
  slice_head(n = 5)

top_labels <- rbind(top_upregulated, top_downregulated)
top_labels$gene <- rownames(top_labels)
```

```{r}
##### Creating Volcano Plot ####
library(ggplot2)
library(ggrepel)

#plot log2FC against -log10(adjusted p value)
volcano_plot <- ggplot(res, aes(x = log2FoldChange, y = -log10(padj), color = Expression)) +
    geom_vline(xintercept = c(-4,4), col = "gray", linetype = "dashed")+ #log2FC threshold
    geom_hline(yintercept = c(-log10(0.001)), col = "gray", linetype = "dashed")+ #P-value threshold
    scale_color_manual(values = c("Upregulated" = "salmon", "Downregulated" = "cyan3", "NS" = "gray"))+ #colour upregulated/downregulated
    geom_point(alpha = 0.6, size=1) +
    geom_text_repel(
    data = top_labels,
    aes(x = log2FoldChange, y = -log10(padj), label = gene, color = "black"),
    size = 3, max.overlaps = 10
    ) +
    theme_minimal(base_size = 10)+
    ggtitle('Volcano Plot: Wizard Freezing vs. Wizard Acclimated Freezing')+
    theme(
      plot.margin = unit(c(2.5,1,1,1), "cm"),
      plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
      plot.title.position = "plot") +
    labs(x = "Log2 Fold Change", y = "-Log10 Adjusted P-value")+
    coord_cartesian(clip = "off")

ggsave("volcano_plot_WFvsWAF.png", plot = volcano_plot, width = 8, height = 7, dpi = 300)
```